resourcemanager/util/mock_gpool.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "time"
+
+// MockGPool is only for test
+type MockGPool struct {
```

Comment:
Why do we need mock pool?

Reply:
No, we cannot import real code. it will cause cycle import.

Reply:
It will run for a long time to create different scenes and it will be flaky because it is easy to be affected by the environment.

Reply:
small cases use small tests. large cases use large tests.